torch/_weights_only_unpickler.py

```diff
@@ -162,10 +308,28 @@ def load(self):
             elif key[0] == REDUCE[0]:
                 args = self.stack.pop()
                 func = self.stack[-1]
-                if func not in _get_allowed_globals().values():
+                if (
+                    func not in _get_allowed_globals().values()
+                    and func not in _get_user_allowed_globals().values()
+                ):
                     raise RuntimeError(
                         f"Trying to call reduce for unrecognized function {func}"
                     )
+                # Prevent tensor subclass type that is pushed onto stack in GLOBAL from
+                # being passed as arg anywhere except the second arg of _rebuild_from_type_v2
+                arg_is_subclass_type = [
+                    i
+                    for i, arg in enumerate(args)
+                    if isinstance(arg, type) and arg in self.tensor_subclasses_found
```

Comment:
```suggestion
                    if isinstance(arg, type) and issubclass(arg, torch.Tensor)
```
Why not this?

Reply:
I think we should never let type be on the unpickler's stack

I changed the strategy here to 
(1) `GLOBAL` finds the subclass, if it meets the criteria, it pushes the str "module.attr" onto the unpickler's stack and adds tensor_subclasses_found["module.attr"] = module.attr
(2) when executing REDUCE for `rebuild_from_type_v2`, we properly translate the str back to the actual type

Does that sound reasonable to you?